### PR TITLE
feat: open-source pms

### PR DESCRIPTION
**Pull request by yashvardhan-kukreja**
**Description**

Open-sources the PMS

**Tests**

Please describe any tests you've added. If you've added no tests, or left important behavior untested, please explain why not.

**Additional context**

Most of the changes I made are to the .circleci configs ensuring:
- To push the peer-mgmt-service to `oplabs-tools-artifacts/images` (favoring open-sourcing) as well as `oplabs-tools-artifacts/internal-images` (favoring backwards-compatibility with the previous implementation)

Rest of the very few changes under `peer-mgmt-service` are only related linter issues.

For a more thorough reference, the structure of the commits associated with this PR depict a better picture of these changes.

**Metadata**

- Fixes ethereum-optimism/platforms-team#303

### CODE REVIEW

File: peer-mgmt-service/README.md
Comment (yashvardhan-kukreja) [inline peer-mgmt-service/README.md:7]: Note: made a nitty fix here (from 1.18 to 1.21)

File: .circleci/continue_config.yml
Comment (yashvardhan-kukreja) [inline .circleci/continue_config.yml:465]: Note: made a change here to `oplabs-gcr-release`

File: .circleci/continue_config.yml
Comment (zhwrd) [inline .circleci/continue_config.yml:478]: not sure the backwards compatibility is necessary imo.. we should just do a fast follow image update after we merge this

File: .circleci/continue_config.yml
Comment (zhwrd) [inline .circleci/continue_config.yml]: can we expand the tag name to `peer-mgmt-service` so its consistent with the code and docker image name?

we can keep the abbreviations in circleci configs, but externally it should be consistent

**Review by zhwrd — CHANGES_REQUESTED** [review]
otherwise lgtm

File: peer-mgmt-service/cmd/pms/main.go
Comment (bitwiseguy) [inline peer-mgmt-service/cmd/pms/main.go]: Seems like all imports should point to this local, public repo instead of pulling from the private one

**Comment by bitwiseguy** [comment]
After addressing other comments, would be good to manually test the `Tag release` github action to ensure a new version of the `peer-mgmt-service` image is successfully published and can be pulled, run without failure

File: .circleci/continue_config.yml
Comment (bitwiseguy) [reply .circleci/continue_config.yml:478]: Could go ahead and create a draft pr against the private repo to remove it there. Would be good to remove it shortly after this pr is merged to avoid confusion around multiple sources for pms images

Comment (yashvardhan-kukreja) [reply .circleci/continue_config.yml:478]: > not sure the backwards compatibility is necessary imo.. we should just do a fast follow image update after we merge this

Sure I don't mind, I personally kept it as a safety measure in case we were to potentially release a quick hotfix to PMS in this repo while the Deployments in our clusters still pointed to the private images. (as a failsafe).

> Could go ahead and create a draft pr against the private repo to remove it there. Would be good to remove it shortly after this pr is merged to avoid confusion around multiple sources for pms images

Already done [here](https://github.com/ethereum-optimism/infrastructure-services/pull/308) : ) 

> Would be good to remove it shortly after this pr is merged

Yep, for the same reason, I intentionally commented region in the config dedicated to the backwards-compatibility changes (makes it easy to spot and remove them).

**Comment by bitwiseguy** [comment]
Should update this github action file to add `peer-mgt-service` as an option: https://github.com/ethereum-optimism/infra/blob/main/.github/workflows/tag-service.yml

**Review by bitwiseguy — APPROVED** [review]
LGTM!

File: .circleci/continue_config.yml
Comment (zhwrd) [reply .circleci/continue_config.yml:478]: Yeah makes sense.. anyways I doubt we will have any hotfix / new changes to this any time soon. Will leave it up to you.

**Review by zhwrd — APPROVED** [review]

**Pull request merged**
